src/index.js

```diff
@@ -1,14 +1,125 @@
-import _ from 'lodash';
 import './style.css';
+import Task from './class.js';
 
-function component() {
-  const element = document.createElement('div');
+const form = document.querySelector('.todo-add');
+const input = document.querySelector('.form-field');
+const ul = document.getElementById('todo-list');
+let li;
 
-  // Lodash, now imported by this script
-  element.innerHTML = _.join(['Hello', 'ahmed'], ' ');
-  element.classList.add('hello');
+let tasks = [];
 
-  return element;
+function getTasks() {
+  if (localStorage.getItem('tasks') == null) {
+    tasks = [];
+  } else {
+    tasks = Array.from(JSON.parse(localStorage.getItem('tasks')));
+  }
+  return tasks;
 }
 
-document.body.appendChild(component());
+function prepareEdit(task, btn) {
+  btn.addEventListener('click', () => {
+    const el = document.querySelector(`[data-index="${task.index}"]`);
+    const p = el.children[1];
+    const input = document.createElement('input');
+    input.type = 'text';
+    input.value = p.textContent;
+    input.id = task.index;
+    el.insertBefore(input, p);
+    el.removeChild(p);
+    input.focus();
+    input.select();
+    input.classList.add('edit');
+  });
+}
```

Comment:
- [ ] I like the way you implemented the edit. However, when I edit a task, the edited description is not saved to the localStorage, so when I reload the page, it shows undefined as the value for the description. What you want to do is to edit the value of the description and save the array with the edited value to the localStorage.

_**Hint**_:  _You should filter through the array, select that particular object, set the value of the `description` key of that object to the input value then save the whole array to the local storage._

Reply:
👍🏽 This works fine now